service/controller/chartconfig/v7/resource/chartstatus/create.go

```diff
@@ -29,30 +31,54 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	if key.ReleaseStatus(customObject) != releaseContent.Status {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, releaseContent.Status))
-
-		customObjectCopy := customObject.DeepCopy()
-		customObjectCopy.Status.ReleaseStatus = releaseContent.Status
+	var status, reason string
+	{
+		if key.IsCordoned(customObject) {
+			status = releaseStatusCordoned
+			reason = key.CordonReason(customObject)
+		} else {
+			status = releaseContent.Status
+			if releaseContent.Status != releaseStatusDeployed {
+				releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
+				if err != nil {
+					return microerror.Mask(err)
+				}
 
-		if releaseContent.Status != releaseStatusDeployed {
-			releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
-			if err != nil {
-				return microerror.Mask(err)
+				reason = releaseHistory.Description
 			}
+		}
+	}
 
-			customObjectCopy.Status.Reason = releaseHistory.Description
-		} else {
-			customObjectCopy.Status.Reason = ""
+	chartConfigStatus := v1alpha1.ChartConfigStatus{
+		ReleaseStatus: status,
+		Reason:        reason,
+	}
+
+	if customObject.Status != chartConfigStatus {
+		// Get chartconfig CR again to ensure the resource version is correct.
+		currentCR, err := r.g8sClient.CoreV1alpha1().ChartConfigs(customObject.Namespace).Get(customObject.Name, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if currentCR.Status.Reason != chartConfigStatus.Reason {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting reason for release '%s' to '%s'", releaseName, reason))
+			currentCR.Status.Reason = chartConfigStatus.Reason
 		}
-		_, err := r.g8sClient.CoreV1alpha1().ChartConfigs(customObject.Namespace).UpdateStatus(customObjectCopy)
+
+		if currentCR.Status.ReleaseStatus != chartConfigStatus.ReleaseStatus {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, status))
```

Comment:
I suggested this already in another place. `%#q` is defined in https://github.com/giantswarm/fmt/blob/master/go/log_messages.md#general-guidelines and does the quoting for you. 

```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release %#q status to %#q", releaseName, status))
```